pkg/sql/pg_catalog_test.go

```diff
@@ -170,9 +215,396 @@ func rewriteDiffs(t *testing.T, diffs PGCatalogTables, diffsFile string) {
+				t.Log(err)
+				continue
+			}
+			output.appendString(createTable)
+		}
+	})
+}
+
+// fixPgCatalogGo will update pgCatalog.allTableNames, pgCatalog.tableDefs and will add needed virtualSchemas
+func fixPgCatalogGo(notImplemented PGCatalogTables) {
+	allTableNamesText := getAllTableNamesText(notImplemented)
+	tableDefinitionText := getTableDefinitionsText(pgCatalogGo, notImplemented)
+
+	rewriteFile(pgCatalogGo, func(input *os.File, output outputFile) {
+		reader := bufio.NewScanner(input)
+		for reader.Scan() {
+			text := reader.Text()
+			trimText := strings.TrimSpace(text)
+			if trimText == virtualTablePosition {
+				//VirtualSchemas doesn't have a particular place to start we just print it before virtualTablePosition
+				output.appendString(printVirtualSchemas(notImplemented))
+			}
+			output.appendString(text)
+			output.appendString("\n")
+
+			switch trimText {
+			case tableDefsDeclaration:
+				printBeforeClosure(reader, output, tableDefsClosure, tableDefinitionText)
+			case allTableNamesDeclaration:
+				printBeforeClosure(reader, output, allTableNamesClose, allTableNamesText)
+			}
+		}
+	})
+}
+
+// printBeforeClosure will skip all the lines and print `s` text when finds the closure text
+func printBeforeClosure(reader *bufio.Scanner, output outputFile, closure string, s string) {
+	for reader.Scan() {
+		text := reader.Text()
+		trimText := strings.TrimSpace(text)
+
+		if strings.HasPrefix(trimText, "//") {
+			output.appendString(text)
+			output.appendString("\n")
+			continue
+		}
+		if trimText != closure {
+			continue
+		}
+		output.appendString(s)
+		output.appendString(text)
+		output.appendString("\n")
+		break
+	}
+}
+
+// getPgCatalogConstants reads catconstant and retrieves all the constant with `PgCatalog` prefix
+func getPgCatalogConstants(
+	t *testing.T, inputFileName string, notImplemented PGCatalogTables,
+) []string {
+	pgConstantSet := make(map[string]struct{})
+	f, err := os.Open(inputFileName)
+	if err != nil {
+		t.Logf("Problem getting pgCatalogConstants: %v", err)
+		t.Fatal(err)
+	}
+	defer dClose(f)
+	reader := bufio.NewScanner(f)
+	for reader.Scan() {
+		text := strings.TrimSpace(reader.Text())
+		if strings.HasPrefix(text, pgCatalogPrefix) {
+			if text == pgCatalogIDConstant {
+				continue
+			}
+			pgConstantSet[text] = none
+		}
+	}
+	for tableName := range notImplemented {
+		pgConstantSet[constantName(tableName, tableIDSuffix)] = none
+	}
+	pgConstants := make([]string, 0, len(pgConstantSet))
+	for pgConstantName := range pgConstantSet {
+		pgConstants = append(pgConstants, pgConstantName)
+	}
+	return pgConstants
+}
+
+// outputFile wraps an *os.file to avoid explicit error checks on every WriteString
+type outputFile struct {
+	f *os.File
+}
+
+// appendString calls WriteString and panics on error
+func (o outputFile) appendString(s string) {
+	if _, err := o.f.WriteString(s); err != nil {
+		panic(fmt.Errorf("error while writing string: %s: %v", s, err))
```

Comment:
I think we can pass `t *testing` here and use t.Fatalf instead for a clearer error message

Reply:
And in general we can replace all the panics with this

Reply:
I am going to push back here:
If you notice, the whole purpose of outputFile type is to wrap io.File (for writers) and have a helper function that avoids any linter complain from the error returned by "WriteString". Which is super improbable that it happen and it would not happen in normal situations.

So I strongly believe that the refactoring of this method to receive *testing.T and using it instead of panic, it is a big effort and the benefit is not really worthy.